server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+        missing_params = []
+        units = body.get('units')
+        if units is None:
+            missing_params.append('units')
+        options = body.get('options')
+        if options is None:
+            missing_params.append('options')
+
+        if missing_params:
+            raise MissingValue(missing_params)
+
+        agent_manager = factory.consumer_agent_manager()
+        task = agent_manager.update_content(consumer_id, units, options)
+        raise OperationPostponed(TaskResult.from_task_status_dict(task))
+
+    def uninstall(self, request, consumer_id):
+        """
+        Uninstall content (units) on a consumer.
+
+        Expected body: {units:[], options:<dict>}
+        where unit is: {type_id:<str>, unit_key={}} and the
+        options is a dict of uninstall options.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :raises MissingValue: if some values are missing
+        :raises OperationPostponed: when an async operation is performed.
+        """
+
+        body = request.body_as_json
+        missing_params = []
+        units = body.get('units')
+        if units is None:
+            missing_params.append('units')
+        options = body.get('options')
+        if options is None:
+            missing_params.append('options')
+
+        if missing_params:
+            raise MissingValue(missing_params)
+
+        agent_manager = factory.consumer_agent_manager()
+        task = agent_manager.uninstall_content(consumer_id, units, options)
+        raise OperationPostponed(TaskResult.from_task_status_dict(task))
+
+
+class ConsumerHistoryView(View):
+    """
+    View for consumer history retrieval.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id):
+        """
+        Retrieve histroy for the specified consumer
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :return: Response representing the binding
+        :rtype: django.http.HttpResponse
+        """
+
+        # Check that the consumer exists and raise a MissingResource exception, in case it doesn't.
+        factory.consumer_manager().get_consumer(consumer_id)
+
+        query_param = request.GET
+        filters = _ensure_input_encoding(query_param)
+        event_type = filters.get('event_type', None)
+        limit = filters.get('limit', None)
+        sort = filters.get('sort', None)
+        start_date = filters.get('start_date', None)
+        end_date = filters.get('end_date', None)
+
+        if sort is None:
+            sort = 'descending'
+
+        if limit:
+            try:
+                limit = int(limit)
+            except ValueError:
+                raise InvalidValue('limit')
```

Comment:
i did not, i remember you had same issue, i thought that just referencing to your bug will be enough, since the nature of the bug is the same.But i was not able to find it, can you please paste me the link? @asmacdo